### PR TITLE
Ftr further train

### LINKED ISSUE #50: Add a "further train" mode ?
- "resume_epoch" is used when the pre-defined number of epochs of training isn't completed. A model's state dict and the optimizer will be loaded from a snapshot. config["trainer"]["resume_epoch"] has to be no greater than config["trainer"]["epochs"].
- Fine-tuning is for a new dataset and the optimizer will be re-initialized.
- "further train" is for situations where the finished training is deemed insufficient and we want to train for some more epochs from a snapshot model without re-initializing the optimizer. Basically this is equivalent to fine-tuning but with network_utils.load_epoch() in stead of network_utils.load()

@bohaohuang let me know if this makes sense to you.

### PR DESCRIPTION
**Pull request by bohaohuang**
This is for #50 
To enable "further train", set trainer->further_train to True, then when loading the model, optimizer will also be loaded. 
Note: further train mode is finetune model plus setting further_train to True

### CODE REVIEW

**Review by waynehuu — APPROVED** [review]

File: network/network_utils.py [hunk 1]
```diff
@@ -217,6 +227,9 @@ def load(model, model_path, relax_load=False, disable_parallel=False):
         # FIXME this is a adhoc fix to be compatible with RSMoCo
         pretrained_state = flex_load(model.state_dict(), checkpoint['model'], relax_load, disable_parallel)
         model.load_state_dict(pretrained_state, strict=False)
+    if optm is not None:
+        assert device is not None
+        load_optim(optm, checkpoint['opt_dict'], device)
```
Comment (waynehuu) [inline network/network_utils.py:232]: @bohaohuang will load_optim here be affected by the multi-gpu fixing if I put it this way?

**Comment by waynehuu** [comment]
Fixes #51
